bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java

```diff
@@ -163,7 +168,7 @@ public Table getTable(TableName tableName, ExecutorService pool) throws IOExcept
       pool = BigtableSessionSharedThreadPools.getInstance().getBatchThreadPool();
     }
     BatchExecutor batchExecutor = new BatchExecutor(
-         new AsyncExecutor(client, heapSizeManager),
+         new AsyncExecutor(client, new RpcThrottler(resourceLimiter)),
```

Comment:
Perhaps each individual call to .batch should have it's own RpcThrottler (or FlushManager?).  There could be multiple concurrent batches happening for a given BatchExecutor, and each batch should be handled by itself with separate flush management per batch.